py/src/nupic/examples/hotgym.py

```diff
@@ -0,0 +1,161 @@
+ 'time': {'timeOfDay': (30, 2), 'weekend': 29},
+ 'tm': {'activationThreshold': 17,
+        'cellsPerColumn': 27,
+        'initialPerm': 0.20991245171830633,
+        'maxSegmentsPerCell': 152,
+        'maxSynapsesPerSegment': 25,
+        'minThreshold': 7,
+        'newSynapseCount': 20,
+        'permanenceDec': 0.09335662142342668,
+        'permanenceInc': 0.08673586267418514}}
+
+
+def main(parameters=default_parameters, argv=None, verbose=True):
+
+  timeOfDayEncoder = DateEncoder(parameters["time"]["timeOfDay"])
+  weekendEncoder   = DateEncoder(parameters["time"]["weekend"])
+
+  rdseParams = RDSE_Parameters()
+  rdseParams.size       = parameters["enc"]["size"]
+  rdseParams.sparsity   = parameters["enc"]["sparsity"]
+  rdseParams.resolution = parameters["enc"]["resolution"]
+  scalarEncoder = RDSE( rdseParams )
+
+  encodingWidth = (timeOfDayEncoder.size
+                   + weekendEncoder.size
+                   + scalarEncoder.size)
+  enc_info = Metrics( [encodingWidth], 999999999 )
+
+  spParams = parameters["sp"]
+  sp = SpatialPooler(
+    inputDimensions            = (encodingWidth,),
+    columnDimensions           = (spParams["columnCount"],),
+    potentialPct               = spParams["potentialPct"],
+    potentialRadius            = encodingWidth,
+    globalInhibition           = True,
+    localAreaDensity           = -1,
+    numActiveColumnsPerInhArea = spParams["numActiveColumnsPerInhArea"],
+    synPermInactiveDec         = spParams["synPermInactiveDec"],
+    synPermActiveInc           = spParams["synPermActiveInc"],
+    synPermConnected           = spParams["synPermConnected"],
+    boostStrength              = spParams["boostStrength"],
+    wrapAround                 = True
+  )
+  sp_info = Metrics( sp.getColumnDimensions(), 999999999 )
+
+  tmParams = parameters["tm"]
+  tm = TemporalMemory(
+    columnDimensions          = (spParams["columnCount"],),
+    cellsPerColumn            = tmParams["cellsPerColumn"],
+    activationThreshold       = tmParams["activationThreshold"],
+    initialPermanence         = tmParams["initialPerm"],
+    connectedPermanence       = spParams["synPermConnected"],
+    minThreshold              = tmParams["minThreshold"],
+    maxNewSynapseCount        = tmParams["newSynapseCount"],
+    permanenceIncrement       = tmParams["permanenceInc"],
+    permanenceDecrement       = tmParams["permanenceDec"],
+    predictedSegmentDecrement = 0.0,
+    maxSegmentsPerCell        = tmParams["maxSegmentsPerCell"],
+    maxSynapsesPerSegment     = tmParams["maxSynapsesPerSegment"]
+  )
+  tm_info = Metrics( [tm.numberOfCells()], 999999999 )
+
+  anomaly_history = AnomalyLikelihood()
+
+  # Read the input file.
+  records = []
+  with open(_INPUT_FILE_PATH, "r") as fin:
+    reader = csv.reader(fin)
+    headers = next(reader)
+    next(reader)
+    next(reader)
+    for record in reader:
+      records.append(record)
+  for i in range(100):
+    records.append( random.choice(records) )
```

Comment:
what is this? you apped shuffeled data?

Reply:
I wanted to check that the anomaly goes up when the input data doesn't follow the pattern.  This is important if  you optimization parameters based on anomaly, because there are ways to lower the anomaly by breaking it.

Reply:
oh, for that reason, wouldn't it be better to optimize for regression? (I know anomaly may be slightly different task). Or put a comment please.

Reply:
Yes, I want to optimize this for 1-step ahead predictions.  I have a prototype now but it doesn't work very well yet.